configure.ml

```diff
@@ -698,10 +698,16 @@ let check_for_numlib () =
       die "Num library not installed, required for OCaml 4.06 or later"
     | _   -> cprintf "You have the Num library installed. Good!");
   let zarith,_ = tryrun camlexec.find ["query";"zarith"] in
+  let zarith_version, _ = run camlexec.find ["query"; "zarith"; "-format"; "%v"] in
   match zarith with
   | ""  ->
     die "Zarith library not installed, required"
-  | _   -> cprintf "You have the Zarith library installed. Good!"
+  | _   ->
+    let zarith_version_int = List.map int_of_string (numeric_prefix_list zarith_version) in
+    if zarith_version_int >= [1;10;0] then
+      cprintf "You have the Zarith library %s installed. Good!" zarith_version
+    else
+      die ("Zarith version 1.10 is required, you have " ^ zarith_version)
```

Comment:
There is not a single version of Ubuntu nor Debian which ships such a new zarith in libzarith-ocaml-dev.  The newest (in Ubuntu focal and groovy and Debian bullseye and sid) is 1.9.1.  If 1.9.1 is not new enough, I'd like to request that the switch to zarith be reverted until such time as there is at least *some* Ubuntu/Debian version with the relevant version of zarith, ideally a stable one (Debian stable currently ships 1.7.1).  You can see the newest package versions at https://packages.ubuntu.com/search?keywords=ocaml-zarith&searchon=sourcenames and https://packages.debian.org/search?searchon=names&keywords=libzarith-ocaml

Reply:
Added this question to the next call.

Reply:
Usually we have this requirement on OCaml, as upgrading it is significantly complex; in particular it needs recompilation of all OCaml libs.

For an OCaml library I think the situation is much easier, whoever is uploading the Coq Debian package can also upload an updated `zarith` easily. Or alternatively, it can be vendored easily.